code/modules/urist/modules/GOAI/integrations/interaction_definitions/_shared.dm

```diff
@@ -0,0 +1,36 @@
+/proc/GenerateGenericInteractions(var/atom/owner =  null, var/datum_path = null)
+	if(!owner || !datum_path)
+		return
+
+	var/datum/interactions_holder/holder = GLOB.goai_interaction_holders[owner.type]
+
+	if(!holder)
+		if(ispath(datum_path))
+			holder = new datum_path
```

Comment:
Doesn't this new need a paren? Special case? I hate DM...

Reply:
Nope. If it's a variable that's a path, it's optional. `new` is a weird one and can be called in lots of different weird ways
`var/T = new /obj/machinery()`
`var/obj/machinery/M = new()`
`var/type = /obj/machinery; var/x = new type`

It's weird, but used plenty throughout the codebase already